src/hotspot/share/classfile/classLoaderExt.cpp

```diff
@@ -131,6 +138,34 @@ void ClassLoaderExt::setup_module_paths(JavaThread* current) {
   process_module_table(current, met);
 }
 
+bool ClassLoaderExt::has_jar_suffix(const char* filename) {
+  const char* dot = strrchr(filename, '.');
+  if (dot != nullptr && strcmp(dot + 1, "jar") == 0) {
+    return true;
+  }
+  return false;
+}
+
+void ClassLoaderExt::extract_jar_files_from_path(const char* path, GrowableArray<const char*>* module_paths) {
+  DIR* dirp = os::opendir(path);
+  if (dirp == nullptr && errno == ENOTDIR && has_jar_suffix(path)) {
+    module_paths->append(path);
```

Comment:
Does this handle the case where `path` doesn't exist?

Reply:
If the `path` doesn't exist, `dirp` will be nullptr and it will go to the else case. I think `os::readir` on `nullptr` should return a `nullptr`. To make the code clearer, I've added a `nullptr` check on `dirp` in the else case.